src/nemo-dbus/dbus.cpp

```diff
@@ -115,7 +117,7 @@ QVariant demarshallDBusArgument(const QVariant &val, int depth)
         res =  val.value<QDBusSignature>().signature();
     } else if (type == qMetaTypeId<QDBusUnixFileDescriptor>()) {
         /* Convert QDBusUnixFileDescriptor to int */
-        res =  val.value<QDBusUnixFileDescriptor>().fileDescriptor();
+        res = dup(val.value<QDBusUnixFileDescriptor>().fileDescriptor());
```

Comment:
Maybe this should return a QDBusUnixFileDescriptor instead. A new descriptor that has to be explicitly closed seems likely to accidentally leak.

Reply:
Great, actually I'd prefer that too. Updated.